### PR TITLE
Wait on user before rendering the classifier

### PR DESCRIPTION
**Pull request by eatyourgreens**
Package:
app-project

Adds user.checkCurrent() to the User store, and adds `user.loadingState` flag, which is set to `loading` while we run the initial check for a logged-in user. Displays a 'logging in' message while we check for a user.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

**Comment by shaunanoordin** [comment]
## PR Quick Review

This PR prevents the Classifier page in `app-project` from rendering until the app can determine if the user is fully logged in or not. A _"signing in..."_ placeholder message appears while the app determines the login status of the user.

The goal, presumably, is to prevent users from interacting with the Classifier while in a limbo state (unknown if they're _supposed to be_ a known user or an anonymous volunteer).

General feelings: code changes mostly look good n' logical 👌 

Quick thoughts: (all minor, mostly rambling, definitely non-blocking)
- Should the _"is the user logged in?"_ check appear at the Classifier page, or somewhere higher in the component hierarchy?
- `user.pending` (boolean) makes perfect sense to me as a way to indicate the state of the User data, but I'm wondering if want to stick with the pattern established with `project.loadingState`?
  - Basically, do we ever need to know if the user check errors out, i.e. `if (user.loadingState === asyncStates.error) ...` ?
- If I'm reading the code right, `user.pending` starts off as true - so won't a non-logged in user who's accessing the Classifier see a "Signing in..." message before quickly being informed that "yep, you're definitely not signed in"?
  - Not a bug, just thinking that the messaging might benefit from being changed to _"synchronalising with the user database-matrix"_ or etc

Notes to self:
- For testing, check what happens if you login, logout, and login again. Check if `pending` is reset to true during multiple state transitions.

**Comment by eatyourgreens** [comment]
>* Should the _"is the user logged in?"_ check appear at the Classifier page, or somewhere higher in the component hierarchy?

At the moment, the classifier is the only component that's rendered by client-side JS. I think the HTML for everything else is rendered on the server, before you're logged in anyway.

I don't think we've got any code yet that reacts to changes in user state beyond the initial requests for a logged-in user.

Good point about whether we want an error state in case the browser is offline, or auth isn't available for some reason.

File: packages/app-project/components/ClassifyPage/components/ClassifierWrapper/ClassifierWrapperContainer.js [hunk 1]
```diff
@@ -42,7 +43,15 @@ export default class ClassifierWrapperContainer extends Component {
       )
     }
 
-    if (project.loadingState === 'success') {
+    if (user.pending) {
```
Comment (shaunanoordin) [inline packages/app-project/components/ClassifyPage/components/ClassifierWrapper/ClassifierWrapperContainer.js]: [Major][Bug] `user` here is undefined, I think we need to specify it in...
```
  render () {
    const { authClient, project } = this.props  // ...here
```

Comment (eatyourgreens) [reply packages/app-project/components/ClassifyPage/components/ClassifierWrapper/ClassifierWrapperContainer.js]: D'oh! I really should set up linting for this repo, to catch errors like this.

Comment (eatyourgreens) [reply packages/app-project/components/ClassifyPage/components/ClassifierWrapper/ClassifierWrapperContainer.js]: Rebased to latest master and fixed. You'll probably need to discard your local branch and pull this one down again.

File: packages/app-project/components/ClassifyPage/components/ClassifierWrapper/ClassifierWrapperContainer.js [hunk 2]
```diff
@@ -51,6 +60,14 @@ export default class ClassifierWrapperContainer extends Component {
       )
     }
 
+    if (project.loadingState === asyncStates.error) {
+      return (
+        <p>
```
Comment (rogerhutchings) [inline packages/app-project/components/ClassifyPage/components/ClassifierWrapper/ClassifierWrapperContainer.js]: This could reuse the `ErrorMessage` component - https://github.com/zooniverse/front-end-monorepo/blob/master/packages/app-project/components/ClassifyPage/components/ClassifierWrapper/components/ErrorMessage/ErrorMessage.js

**Comment by eatyourgreens** [comment]
@shaunanoordin I've replaced `user.pending` with `user.loadingState`. You're right, this store should use the same pattern as the others, if only for the sanity of anyone else who has to work with this code.

File: packages/app-project/components/ClassifyPage/components/ClassifierWrapper/ClassifierWrapperContainer.js (hunk 2 above)
Comment (eatyourgreens) [reply packages/app-project/components/ClassifyPage/components/ClassifierWrapper/ClassifierWrapperContainer.js]: Good point.

Comment (eatyourgreens) [reply packages/app-project/components/ClassifyPage/components/ClassifierWrapper/ClassifierWrapperContainer.js]: I've added a simple test which should display project loading errors or classifier errors with the `ErrorMessage` component, I think.

**Comment by shaunanoordin** [comment]
I understand that we're waiting for a discussion to resolve regarding whether #454 or this branch should be merged first, right? In that case, I'll hold of clicking that _squish n' murge_ button.

**Comment by eatyourgreens** [comment]
I think #454 moves this file without changing it, so I’m inclined to merge this first. Happy to wait for an opinion from @rogerhutchings since we are creating conflicts for his PR.

**Comment by rogerhutchings** [comment]
Yup, this can go first

**Comment by eatyourgreens** [comment]
#474 should re-mount the classifier on login/logout, by the simple expedient of keying the classifier component by user ID.
